src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+     * 
+     * @param taskNumber The task number (starting from 1) that was marked as done. 
+     */
+    private static void displayMarkTaskSuccessMessage(int taskNumber) {
+        displayToUser(MESSAGE_MARKED, tasks[taskNumber].toString());
+    }
+
+    /**
+     * Adds a new todo task to task list. 
+     */
+    private static void executeAddTodo() {
+        tasks[sizeOfTaskList] = new Todo(userCommand);
+        sizeOfTaskList++;
+        displayAddTaskSuccessMessage();
+    }
+
+    /**
+     * Adds a new deadline task to task list. 
+     * Processes the user input to extract date. 
+     * If date is not null, create new deadline task. 
+     * Otherwise, display invalid command error. 
+     */
+    private static void executeAddDeadline() {
+        String date = processParameters(DEADLINE_DATA_PREFIX_BY);
+        if (date != null) {
+            tasks[sizeOfTaskList] = new Deadline(userCommand, date);
+            sizeOfTaskList++;
+            displayAddTaskSuccessMessage();
+        } else {
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+        }
+    }
+
+    /**
+     * Adds a new event task to task list. 
+     */
+    private static void executeAddEvent() {
+        // Process event parameters
+        String date = processParameters(EVENT_DATA_PREFIX_AT);
+        if (date != null) {
+            tasks[sizeOfTaskList] = new Event(userCommand, date);
+            sizeOfTaskList++;
+            displayAddTaskSuccessMessage();
+        } else {
+            // Display error message
+            displayToUser(ERROR_INVALID_COMMAND_RECEIVED);
+        }
+
+    }
+
+    /**
+     * Finds the location of the filters and locate the date from the task. 
+     * Removes the task date information from userCommand, keeping only the task name in userCommand. 
+     * 
+     * @param filterString The string to find in userCommand depending on an event (/at) or deadline (/by). 
+     * @return The date extracted from userCommand. 
+     */
+    private static String processParameters(String filterString) {
+        int indexOfDate = userCommand.indexOf(filterString);
+        if (indexOfDate > 0) {
+            String date = userCommand.substring(indexOfDate + 4);
+            userCommand = userCommand.substring(0, indexOfDate - 1);
+            return date;
+        } else {
+            return null;
+        }
+        
+    }
+
+    /**
+     * Displays success message after adding new task. 
+     */
+    private static void displayAddTaskSuccessMessage() {
+        displayToUser(MESSAGE_ADDED, "  " + getTask(sizeOfTaskList-1).toString(), String.format(MESSAGE_NUMBER_OF_TASKS,
+                sizeOfTaskList));
+    }
+
+    /**
+     * Displays a given message to the user. 
+     * 
+     * @param message Message to be displayed. 
+     */
+    private static void displayToUser(String... message) {
+        System.out.println(String.format("\t%s", MESSAGE_BORDER));
+        for (String m : message) {
```

Comment:
Maybe you can consider using 'message' instead of 'm' as the variable name (to make it more descriptive). Also, since the argument (message) is an array, you can rename it as messages so that it is clearer.

Reply:
Will do! Thanks